oap-server/server-receiver-plugin/configuration-discovery-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/recevier/configuration/discovery/AgentConfigurationsWatcher.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.oap.server.recevier.configuration.discovery;
+
+import java.io.StringReader;
+import org.apache.commons.codec.digest.DigestUtils;
+import org.apache.skywalking.oap.server.configuration.api.ConfigChangeWatcher;
+import org.apache.skywalking.oap.server.core.Const;
+import org.apache.skywalking.oap.server.library.module.ModuleProvider;
+
+/**
+ * AgentConfigurationsWatcher used to handle dynamic configuration changes.
+ */
+public class AgentConfigurationsWatcher extends ConfigChangeWatcher {
+    private volatile String settingsString;
+    private volatile AgentConfigurationsTable agentConfigurationsTable;
+    /**
+     * The uuid is based on calculation by md5.
+     */
+    private volatile String uuid;
```

Comment:
UUID should target to the specific service's configuration. Otherwise, one config of one service changed, would trigger the global agent config updates.

Reply:
done.